cirq-core/cirq/transformers/routing/line_initial_mapper.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Concrete implementation of AbstractInitialMapper that places lines of qubits onto the device."""
+
+from typing import Dict, Optional, TYPE_CHECKING
+import networkx as nx
+
+from cirq import value, _compat
+from cirq.transformers import routing
```

Comment:
Don't import `routing` since it can lead to circular imports (routing imports `line_initial_mapper` and `line_initial_mapper.py` imports `routing`). 
Import the specific submodule instead (`from cirq.transformers.routing import initial_mapper`)